src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -22,27 +22,27 @@ internal partial class Checksum
         private static readonly ObjectPool<IncrementalHash> s_incrementalHashPool =
             new ObjectPool<IncrementalHash>(() => IncrementalHash.CreateHash(HashAlgorithmName.SHA256), size: 20);
 
-        public static Checksum Create(string val)
+        public static Checksum Create(IEnumerable<string> values)
         {
             using var pooledHash = s_incrementalHashPool.GetPooledObject();
             using var pooledBuffer = SharedPools.ByteArray.GetPooledObject();
             var hash = pooledHash.Object;
 
-            var stringBytes = MemoryMarshal.AsBytes(val.AsSpan());
-            Debug.Assert(stringBytes.Length == val.Length * 2);
-            var buffer = pooledBuffer.Object;
-
-            var index = 0;
-            while (index < stringBytes.Length)
+            foreach (var value in values)
             {
-                var remaining = stringBytes.Length - index;
-                var toCopy = Math.Min(remaining, buffer.Length);
+                AppendData(hash, pooledBuffer.Object, value);
```

Comment:
There needs to be a write of a null or something between each line or otherwise the hash of "AB" and "A", "B" will be the same.

Reply:
I guess semicolon still scares me here since that's a character we might end up using if this function gets reused. Just do the null character to be safe?